### PR TITLE
EntityDebugDisplayBus SetColor Explicit Naming

### LINKED ISSUE #8807: DebugDisplayRequests EBus Uses Overloaded Function
**Describe the bug**
I was using DebugDisplayRequests::SetColor and the compiler couldn't figure out which overloaded function to event broadcast. The  SetColor methods should be broken into 3 separate methods: 
  1. SetColor(AZ::Color) 
  2. SetColorF(float r,g,b,a)
  3. SetColorFromVector4(Vector4 color)? 

Also that would follow Lumberyard standards... https://docs.aws.amazon.com/lumberyard/latest/userguide/ebus-usage-and-examples.html
_"Use descriptive names in EBuses, and avoid overloaded functions. Explicit and descriptive function names prevent future API name collisions when classes inherit your EBus interfaces. Avoiding overloaded functions improves the experience of using your EBuses. This is especially true from scripting environments such as Lua, in which descriptive names improve readability and clarity."_

### PR DESCRIPTION
**Pull request by AMZN-Gene**
Changed DebugDisplayRequests's 3 SetColor overloaded functions to 1) SetColor(AZ::Color), 2) SetColorF(float r,g,b,a), and 3) SetColorFromVector4(Vector4 color).

https://docs.aws.amazon.com/lumberyard/latest/userguide/ebus-usage-and-examples.html
"Use descriptive names in EBuses, and avoid overloaded functions. Explicit and descriptive function names prevent future API name collisions when classes inherit your EBus interfaces. Avoiding overloaded functions improves the experience of using your EBuses. This is especially true from scripting environments such as Lua, in which descriptive names improve readability and clarity."

Tested: Using the new bus color call for Multiplayer Editor

Change made based on conversation: https://discord.com/channels/805939474655346758/816043601754325023/962044821869051964
Fixes #8807

### CODE REVIEW

**Comment by AMZN-Olex** [comment]
This is great, thanks!

File: Gems/AtomLyIntegration/AtomBridge/Code/Source/AtomDebugDisplayViewportInterface.h [hunk 1]
```diff
@@ -135,9 +135,9 @@ namespace AZ::AtomBridge
         // Commented out function prototypes are remaining part of the api
         // waiting to be implemented.
         // work tracked in [ATOM-3459]
-        void SetColor(float r, float g, float b, float a = 1.f) override;
+        void SetColorF(float r, float g, float b, float a = 1.f) override;
         void SetColor(const AZ::Color& color) override;
-        void SetColor(const AZ::Vector4& color) override;
+        void SetColorFromVector4(const AZ::Vector4& color) override;
```
[the remark this answers is not in the document]
Comment (AMZN-Gene) [reply Gems/AtomLyIntegration/AtomBridge/Code/Source/AtomDebugDisplayViewportInterface.h]: Fixed!

**Comment by AMZN-Gene** [comment]
> This change looks fine. However, if the functions and call sites are being updated anyway then I propose getting rid of all of the overloads and just having the one set color function. AZ::Color has an extensive list of constructors and create functions that make the overloads unnecessary.

fixed!

**Review by gadams3 — APPROVED** [review]

**Review by hultonha — APPROVED** [review]
Nice! Thanks for updating 👍

**Pull request merged**